test/test_ngraph_exec.cpp

```diff
@@ -0,0 +1,499 @@
+      try {
+        outputs[i] = ng_exec->create_output_tensor(pipelined_output_indexes[i],
+                                                   pipeline_depth);
+
+      } catch (const std::exception& exp) {
+        return errors::Internal(
+            "Failed to create pipelined tensor output for index i ", i,
+            ".Got ngraph exception ", exp.what());
+      }
+    }
+
+    io_tensors = {inputs, outputs};
+    return Status::OK();
+  }
+
+  // Creates the tensor from backend of the same shape and data type as TF
+  // Tensor
+  // Copies data from TF Tensor to it
+  Status CreateTensorFromBackend(
+      const shared_ptr<ngraph::runtime::Backend>& ng_backend, Tensor& tf_tensor,
+      shared_ptr<ngraph::runtime::Tensor>& ng_tensor) {
+    ng::element::Type ng_element_type;
+    TF_RETURN_IF_ERROR(
+        TFDataTypeToNGraphElementType(tf_tensor.dtype(), &ng_element_type));
+    ng::Shape ng_shape(tf_tensor.shape().dims());
+    for (int j = 0; j < tf_tensor.shape().dims(); ++j) {
+      ng_shape[j] = tf_tensor.shape().dim_size(j);
+    }
+    TF_RETURN_IF_ERROR(CreateTensorFromBackend(ng_backend, ng_element_type,
+                                               ng_shape, ng_tensor));
+    WriteNGTensor(ng_tensor, &tf_tensor);
+    return Status::OK();
+  }
+
+  // Creates the tensor from backend of the provided shape and data type
+  Status CreateTensorFromBackend(
+      const shared_ptr<ngraph::runtime::Backend>& ng_backend,
+      const ng::element::Type& ng_element_type, const ng::Shape& ng_shape,
+      shared_ptr<ngraph::runtime::Tensor>& ng_tensor) {
+    try {
+      ng_tensor = ng_backend->create_tensor(ng_element_type, ng_shape);
+    } catch (const std::exception& exp) {
+      return errors::Internal(
+          "Failed to create tensor from backend. Got ngraph exception ",
+          exp.what());
+    }
+    return Status::OK();
+  }
+
+  // Executes the graph on TF
+  Status RunGraphOnTF(const Graph& graph,
+                      const vector<pair<string, Tensor>>& feed_dict,
+                      const vector<string>& out_node_names,
+                      vector<Tensor>& out_tensors) {
+    // Create Session
+    SessionOptions options;
+    options.config.mutable_graph_options()
+        ->mutable_optimizer_options()
+        ->set_opt_level(tf::OptimizerOptions_Level_L0);
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_constant_folding(tf::RewriterConfig::OFF);
+    std::unique_ptr<Session> session(NewSession(options));
+
+    // Attach Graph
+    GraphDef gdef;
+    graph.ToGraphDef(&gdef);
+    TF_RETURN_IF_ERROR(session->Create(gdef));
+    DeactivateNGraph();
+    Status status = session->Run(feed_dict, out_node_names, {}, &out_tensors);
+    ActivateNGraph();
+    return status;
+  }
+};
+
+TEST_F(NGraphExecTest, Axpy) {
+  Graph input_graph(OpRegistry::Global());
+  ASSERT_OK(LoadGraph("test_axpy_launchop.pbtxt", &input_graph));
```

Comment:
I think there is a lot of common code between  these tests, and It can be moved to SetUp().

Reply:
I have not looked too much into the existing tests Axpy and Axpy8bit. Changing them are beyond the scope of this PR. It is showing as major change mainly due to renaming of the file